backend/compact-connect/docs/design/README.md

```diff
@@ -31,6 +57,82 @@ efficient processing. A lambda receives messages from the SQS queue. Each messag
 ingested. The lambda receives the data and creates or updates a corresponding license record in the DynamoDB license
 data table.
 
-
 ### Asynchronous validation feedback
 Asynchronous validation feedback for boards to review is not yet implemented.
+
+## User Architecture
+[Back to top](#backend-design)
+
+Authentication with the CompactConnect backend will be controlled through Oauth2 via
+[AWS Cognito User Pools](https://github.com/csg-org/CompactConnect). Clients will be divided into two groups, each
+represented by an independent User Pool: [Staff Users](#staff-users) and [Licensee Users](#licensee-users). See
+the accompanying [architecture diagram](./users-arch-diagram.pdf) for an illustration.
+
+### Staff Users
+
+Staff users come with a variety of different permissions, depending on their role. There are Compact Executive
+Directors, Compact ED Staff, Board Executive Directors, Board ED Staff, and CSG Admins, each with different levels
+of ability to read and write data, and to administrate users. Read permissions are granted to a user for an entire
+compact or not at all. Data writing and user administration permissions can each be granted to a user per
+compact/jurisdiction combination. All of a compact user's permissions are stored in a DynamoDB record that is associated
+with their own Cognito user id. That record will be used to generate scopes in the Oauth2 token issued to them on login.
+See [Implementation of scopes](#implementation-of-scopes) for a detailed explanation of the design for exactly how
+permissions will be represented by scopes in an access token. See
+[Implementation of permissions](#implementation-of-permissions) for a detailed explanation of the design for exactly
+how permissions are stored and translated into scopes.
+
+#### Compact Executive Directors and Staff
+
+Compact ED level staff can have permission to read all compact data as well as to create and manage users and their
+permissions. They can grant other users the ability to write data for a particular jurisdiction and to create more
+users associated with a particular jurisdiction. They can also delete any user within their compact, so long as that
+user does not have permissions associated with a different compact.
```

Comment:
Something to test, perhaps?

Reply:
Yes, we should absolutely test these things, when we get to the user management feature(s). We haven't taken that on yet.